Objects/listobject.c

```diff
@@ -656,6 +656,18 @@ list_item(PyObject *aa, Py_ssize_t i)
         return NULL;
     }
     PyObject *item;
+#ifdef Py_GIL_DISABLED
+    PyObject **ob_item = _Py_atomic_load_ptr(&a->ob_item);
+    item = _Py_atomic_load_ptr(&ob_item[i]);
+    if (item && _Py_TryIncrefCompare(&ob_item[i], item)) {
+        if (ob_item != _Py_atomic_load_ptr(&a->ob_item)) {
+            Py_DECREF(item);
+            goto retry;
+        }
+        return item;
```

Comment:
Don't we have to check `_PyObject_GC_IS_SHARED` at this moment for setting `_PyObject_GC_SET_SHARED` if the object is not owned by the current thread?